pyproject.toml

```diff
@@ -16,6 +16,10 @@ python-dateutil = "^2.9.0.post0"
 setuptools = "*"
 uritemplate = "^4.1.1"
 urllib3 = "^2.2.2"
+requests = "^2.32.3"
+typeguard = "^4.3.0"
+xmlasdict = "^0.2.2"
```

Comment:
xmlasdict dependency is also not required for subyt use - only when you want to deal with xml inputs (which we do, but only inside the tests?)

caveat -- documentation will need to make clear that such usage declares its own dependency on xmlasdict

makes me think (new issue?) we could also register a fake source class that makes that statement explicitely when people try inputtinh files that do require the XMLSource

Reply:
note: currently there is a log.warn already about the xml support being not available when the xmlasdict dependency is not met -- as suggested above, we might extend that policy with a more upfront message / fail when XML processing is actually attempted later

Reply:
Further handling done by #68